include/uam/workers/yr_worker.h

```diff
@@ -0,0 +1,299 @@
+   * @brief
+   *
+   * @param raw_reply
+   * @return
+   */
+  bool processByHandler(const Reply& raw_reply)
+  {
+    auto reply = process(raw_reply);
+    if (!reply)
+      return false;
+
+    if (callback_)
+    {
+      callback_(*reply);
+    }
+    return true;
+  }
+
+protected:
+  /**
+   * @bief Convert value to Hexadecimal in string form
+   *
+   * @param[in] value - Value to be converted
+   * @param[in] size - The number of bytes of the value
+   * @return The hexadecimal value in string form
+   */
+  template <typename T>
+  static std::string toHexString(const T& value, const size_t& size = sizeof(T))
+  {
+    std::stringstream stream;
+    stream << std::uppercase << std::setfill('0') << std::setw(size) << std::hex << value;
+    return stream.str();
+  }
+
+  static std::string encodeCommand(const Request& request)
+  {
+    std::string encoded_request;
+    encoded_request.reserve(sizeof(Request));
+    encoded_request += request.header.stx;
+    encoded_request += toHexString(request.header.cmd_size);
+    encoded_request += request.header.header[0];
+    encoded_request += request.header.header[1];
+    encoded_request += toHexString(request.header.area_type);
+    encoded_request += toHexString(request.header.area_number);
+    encoded_request += toHexString(request.header.start_step);
+    encoded_request += toHexString(request.header.end_step);
+    encoded_request += toHexString(request.header.resolution);
+    encoded_request += toHexString(request.footer.crc);
+    encoded_request += request.footer.etx;
+    return encoded_request;
+  }
+
+  uint16_t calculateCrc(const Request& msg) const
+  {
+    // CRC Calculation
+    // create a tmp string of cmd and header
+    std::string size_cmd_str = toHexString(msg.header.cmd_size) + msg.header.header[0] + msg.header.header[1] +
+                               toHexString(msg.header.area_type) + toHexString(msg.header.area_number) +
+                               toHexString(msg.header.start_step) + toHexString(msg.header.end_step) +
+                               toHexString(msg.header.resolution);
+    return calculateCrc(size_cmd_str.data(), size_cmd_str.size());
+  }
+
+  uint16_t calculateReplyCrc(const Reply& message) const
+  {
+    // validate crc with expected crc:
+    const auto crc_buffer_size =
+      sizeof(Reply) - (sizeof(protocol::CommandReplyHeader::stx) + sizeof(protocol::CommandFooter));
+    return calculateCrc(
+      reinterpret_cast<const char*>(&message) + sizeof(protocol::CommandReplyHeader::stx),
+      crc_buffer_size);
+  }
+
+  /**
+   * @brief Calculate crc
+   *
+   * CRC Standard: Kermit
+   * Polynomial: 0x1021
+   * Shift Direction: Right
+   * Initial Value: 0x0000
+   * Byte Swap: Yes
+   * Reverse CRC Result: Yes
+   *
+   * @param[in] buffer - Buffer
+   * @param[in] byte_count - Number of bytes in the buffer to use
+   * @return checksum
+   */
+  uint16_t calculateCrc(const char* buffer, const std::size_t& byte_count) const
```

Comment:
I might have miss something but the calculation of the CRC in here seems very similar to the one in the `uam_worker_base.h` . The polynomial and remaining configurations are identical.
Could this be generic for both cases? Not required, but for maintenance might be better

Reply:
Moved this function